.repo/sample-list-generator/src/libs/api-detector.ts

```diff
@@ -0,0 +1,161 @@
+import {
+  ApiItem,
+  ApiItemWithType,
+  ApiTypeResult,
+  ExtensionApiMap
+} from '../types';
+import * as babel from '@babel/core';
+import { isIdentifier } from '@babel/types';
+import fs from 'fs/promises';
+import { getAllFiles } from '../utils/filesystem';
+import { loadExtensionApis } from './api-loader';
+
+let EXTENSION_API_MAP: ExtensionApiMap = loadExtensionApis();
+
+export const getApiType = (
+  namespace: string,
+  propertyName: string
+): ApiTypeResult => {
+  namespace = namespace.replace(/_/g, '.');
+
+  if (EXTENSION_API_MAP[namespace]) {
+    const apiTypes = EXTENSION_API_MAP[namespace];
+
+    if (apiTypes.methods.includes(propertyName)) {
+      return 'method';
+    }
+    if (apiTypes.events.includes(propertyName)) {
+      return 'event';
+    }
+    if (apiTypes.properties.includes(propertyName)) {
+      return 'property';
+    }
+    if (apiTypes.types.includes(propertyName)) {
+      return 'type';
+    }
+  }
+  return 'unknown';
+};
+
+export const getApiListForSample = async (
+  sampleFolderPath: string
+): Promise<ApiItemWithType[]> => {
+  // get all js files in the folder
+  const jsFiles = (await getAllFiles(sampleFolderPath)).filter((file) =>
+    file.endsWith('.js')
+  );
+
+  const calls: ApiItemWithType[] = [];
+
+  const parallelHandler = jsFiles.map(async (file) => {
```

Comment:
```suggestion
  await Promise.all(jsFiles.map(async (file) => {
```

I think something like this probably reads well enough and avoids creating a temporary variable?